### ISSUE TITLE
Clarifications on code

### ISSUE THREAD

**Issue by rafaelvalle** [igul222/improved_wgan_training#1]
What's the output of `Discriminator(interpolates)[0]` on the code below in [gan_language.py](https://github.com/igul222/improved_wgan_training/blob/master/gan_language.py#L106)? 
Knowing that `gen_cost = -tf.reduce_mean(Discriminator(fake_inputs))`, I assume that `Discriminator(interpolates)` returns the discriminator evaluation's of the first batch in the interpolates, though this doesn't seem to make any sense.
  
```python
gradients = tf.gradients(Discriminator(interpolates)[0], [interpolates])[0]  
slopes = tf.sqrt(tf.reduce_sum(tf.square(gradients), reduction_indices=[1,2]))  
gradient_penalty = tf.reduce_mean((slopes-1.)**2)  
disc_cost += LAMBDA*gradient_penalty  
```

**Comment by igul222** [comment]
Good catch! We missed this while cleaning up the code for release, it should just read `gradients = tf.gradients(Discriminator(interpolates), [interpolates])[0]`

**Comment by rafaelvalle** [comment]
I'm porting this code to Lasagne and Theano.

Would `gradients = tf.gradients(Discriminator(interpolates), [interpolates])[0]` be equal to `theano.grad(Discriminator(interpolates).sum(), interpolates)`? theano grads expects a scalar and I assume that in your code `Discriminator(interpolates)` returns a vector with the valuations of the discriminator for each interpolation.
